components/contentblocks/indicator-chart/DashboardIndicatorLineChartBlock.tsx

```diff
@@ -0,0 +1,263 @@
+'use client';
+
+import React from 'react';
+import { useTheme } from 'styled-components';
+import { LineChart, ScatterChart } from 'echarts/charts';
+import {
+  GridComponent,
+  TooltipComponent,
+  LegendComponent,
+} from 'echarts/components';
+import * as echarts from 'echarts/core';
+import Chart, { ECOption } from '@/components/paths/graphs/Chart';
+import { DashboardIndicatorLineChartBlock as TDashboardIndicatorLineChartBlock } from '@/common/__generated__/graphql';
+import { getDefaultColors } from './indicator-chart-colors';
+import { linearRegression } from 'common/math';
+
+interface GraphsTheme {
+  categoryColors?: string[];
+  totalLineColor?: string;
+  trendLineColor?: string;
+  goalLineColors?: string[];
+  showTrendline?: boolean;
+}
+
+echarts.use([
+  LineChart,
+  ScatterChart,
+  GridComponent,
+  TooltipComponent,
+  LegendComponent,
+]);
+
+type Props = TDashboardIndicatorLineChartBlock;
+
+const X_SYMBOL =
+  'path://M0.979266 20.7782C-0.192306 21.9497 -0.192307 23.8492 0.979266 25.0208C2.15084 26.1924 4.05033 26.1924 5.22191 ' +
+  '25.0208L13.0001 17.2426L20.7783 25.0208C21.9498 26.1924 23.8493 26.1924 25.0209 25.0208C26.1925 23.8492 26.1925 21.9497 ' +
+  '25.0209 20.7782L17.2427 13L25.0209 5.22181C26.1925 4.05024 26.1925 2.15075 25.0209 0.979174C23.8493 -0.192399 21.9498 ' +
+  '-0.192399 20.7783 0.979174L13.0001 8.75735L5.22191 0.979175C4.05033 -0.192398 2.15084 -0.192398 0.979266 0.979175C-0.192307 ' +
+  '2.15075 -0.192307 4.05024 0.979266 5.22182L8.75744 13L0.979266 20.7782Z';
+
+const DashboardIndicatorLineChartBlock: React.FC<Props> = ({
+  chartSeries,
+  indicator,
+  dimension,
+  showTotalLine,
+}) => {
+  const theme = useTheme();
+  const graphsTheme: GraphsTheme = theme.settings?.graphs ?? {};
+  const unit = indicator?.unit?.name ?? '';
+  const palette = graphsTheme.categoryColors ?? getDefaultColors(theme);
+
+  if (!chartSeries?.length) {
+    return <div>No data</div>;
+  }
+
+  const dimCategoryMap = new Map<
```

Comment:
There's a lot going on in this component, I think we could split it up into smaller functions (`getDimSeries`, `getTotalSeries` etc). But we can leave that for V2 and the other graphs.